examples/go-tracepoint-counter/main.go

```diff
@@ -0,0 +1,152 @@
+func main() {
+	// pull the BPFD config management data to determine if we're running on a
+	// system with BPFD available.
+	paramData, err := configMgmt.ParseParamData(configMgmt.ProgTypeTracepoint, DefaultConfigPath, DefaultByteCodeFile)
+	if err != nil {
+		log.Printf("error processing parameters: %v\n", err)
+		return
+	}
+
+	// determine the path to the tracepoint_stats_map, whether provided via CRD
+	// or BPFD or otherwise.
+	var mapPath string
+	if paramData.CrdFlag { // get the map path from the API resource if on k8s
+		mapPath, err = bpfdAppClient.GetMapPathDyn(BpfProgramConfigName, BpfProgramMapIndex)
+		if err != nil {
+			log.Fatalf("error reading BpfProgram CRD: %v\n", err)
+		}
+	} else { // if not on k8s, find the map path from the system
+
+		// if the bytecode src is not a UUID provided by BPFD, we'll need to
+		// load the program ourselves
+		if paramData.BytecodeSrc != configMgmt.SrcUuid {
+			cleanup, err := loadProgram(&paramData)
+			if err != nil {
+				log.Fatalf("failed to load BPF program: %v\n", err)
+			}
+			defer cleanup(paramData.Uuid)
+		}
+
+		mapPath = fmt.Sprintf("%s/%s/tracepoint_stats_map", DefaultMapDir, paramData.Uuid)
+	}
+
+	// load the pinned stats map which is keeping count of kill -SIGUSR1 calls
+	opts := &ebpf.LoadPinOptions{
+		ReadOnly:  false,
+		WriteOnly: false,
+		Flags:     0,
+	}
+	statsMap, err := ebpf.LoadPinnedMap(mapPath, opts)
+	if err != nil {
+		log.Printf("Failed to load pinned Map: %s\n", mapPath)
+		log.Print(err)
+		return
+	}
+
+	// send a SIGUSR1 signal to this program on repeat, which the BPF program
+	// will report on to the stats map.
+	go func() {
+		for {
+			syscall.Kill(os.Getpid(), syscall.SIGUSR1)
+			time.Sleep(time.Second * 1)
+		}
+	}()
+
+	// retrieve and report on the number of kill -SIGUSR1 calls
+	index := uint32(0)
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+	for range ticker.C {
+		var stats []Stats
+		var totalCalls uint64
+
+		if err := statsMap.Lookup(&index, &stats); err != nil {
+			log.Fatalf("map lookup failed: %v\n", err)
+		}
+
+		for _, stat := range stats {
+			totalCalls += stat.Calls
+		}
+
+		log.Printf("%d SIGUSR1 signals seen\n", totalCalls)
+	}
+}
+
+func loadProgram(paramData *configMgmt.ParameterData) (func(string), error) {
+	// get the BPFD TLS credentials
+	configFileData := configMgmt.LoadConfig(DefaultConfigPath)
+	creds, err := configMgmt.LoadTLSCredentials(configFileData.Tls)
+	if err != nil {
+		return nil, err
+	}
+
+	// connect to the BPFD server
+	ctx := context.Background()
+	conn, err := grpc.DialContext(ctx, "localhost:50051", grpc.WithTransportCredentials(creds))
+	if err != nil {
+		return nil, err
+	}
+	c := gobpfd.NewLoaderClient(conn)
+
+	// create a request to load the BPF program
+	loadRequest := &gobpfd.LoadRequest{
+		Location:    paramData.BytecodeLocation,
+		SectionName: "stats",
+		ProgramType: gobpfd.ProgramType_TRACEPOINT,
+		AttachType:  &gobpfd.LoadRequest_SingleAttach{},
+	}
```

Comment:
Section name is wrong, and the tracepoint name that you want to attach to needs to be filled out:
i.e
```suggestion
	loadRequest := &gobpfd.LoadRequest{
		Location:    paramData.BytecodeLocation,
		SectionName: "tracepoint_kill_recorder",
		ProgramType: gobpfd.ProgramType_TRACEPOINT,
		AttachType: &gobpfd.LoadRequest_SingleAttach{
			SingleAttach: &gobpfd.SingleAttach{Name: "syscalls/sys_exit_kill"},
		},
	}
```